presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1793,9 +1793,39 @@ public OptionalLong metadataDelete(ConnectorSession session, ConnectorTableHandl
     }
 
     @Override
-    public boolean supportsMetadataDelete(ConnectorSession session, ConnectorTableHandle tableHandle)
+    public boolean supportsMetadataDelete(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<ConnectorTableLayoutHandle> tableLayoutHandle)
     {
-        return true;
```

Comment:
Can we have an enumeration of comments to explain the cases when metadata delete can happen and when it can't? By looking at the code, it seems that it's not an all or nothing call given the filter could be "simple" or very "complicated" but turning to be simple if we do constant folding or even evaluation. Of course, there is a limit we can push. We can only inference the delete if the filter is not "that complicated" .

Reply:
@highker James, thank you review.

Strictly speaking we should be able to support metadata delete for any filter that doesn't touch non-partition columns. Today we only support it for range filters on partition columns though (regardless of whether filter pushdown is on or off). I'll add a comment and a TODO.

```
// Allow metadata delete for range filters on partition columns.
// TODO Add support for metadata delete for any filter on partition columns.
```